gsa/src/web/components/form/multiselect.js

```diff
@@ -144,12 +145,12 @@ class MultiSelect extends React.Component {
     this.notifyChange(copy);
   }
 
-  componentWillReceiveProps(nextProps) {
-    const {value} = nextProps;
-
-    if (isDefined(value) && !arraysEqual(value, this.state.selectedItems)) {
-      this.setState({selectedItems: value});
+  static getDerivedStateFromProps(props, state) {
```

Comment:
I am really not sure if using MultiSelect uncontrolled still works now. Maybe we should just remove supporting the uncontrolled case and drop this completely?

Reply:
This is basically the test case "Should change displayed values" in the multiselect tests. The test passes so it should still work. But I am not sure if this is actually something that happens in GSA. If it never happens it should be safe to remove it. Everything else seems to be working fine without getDerivedStateFromProps.

Reply:
I just had a quick look at the MultiSelect code and I would love to remove the internal state and just use the props.value as selected items. If we only provide the controlled mode (by removing the internal state selectedItems) we can just drop the cwrp/getDerivedStateFromProps method.